cmake/Corrosion.cmake

```diff
@@ -1397,3 +1397,74 @@ function(corrosion_install)
         message(FATAL_ERROR "install(EXPORT ...) not yet implemented")
     endif()
 endfunction()
+
+# Create a library target that will run cxxbridge on the given files on a crate
+# that is assumed to contain a module with #[cxx::bridge]
+# Call with corrosion_cxxbridge(<target> CRATE <crate> PATH <tomlPath> FILES [<filePath>, ...])
+# <target> is the name of the target created that will host the cxx interface
+# CRATE <crate> is the target name of the specific target generated by corrosion_import_crate
+# PATH <tomlPath> is the relative path from where this function is called to Cargo.toml of <crate>
+# FILES [<filePath, ...] is the relative paths to files with #[cxx::bridge]. These are relative to <tomlPath>
+# 
+# Example: corrosion_cxxbridge(myCxxTarget CRATE someCrate PATH someCratePath FILES src/lib.rs)
+function(corrosion_cxxbridge target)
+    # This function is based on the details given by https://cxx.rs/build/other.html#producing-the-generated-code
+    # corrosion_cxxbridge requires the cxxbridge executable to be installed
+    # run `cargo install cxxbridge-cmd` if missing
+    find_program(CXXBRIDGE cxxbridge PATHS "$ENV{HOME}/.cargo/bin/" REQUIRED)
+
+    set(OPTIONS)
+    set(ONE_VALUE_KEYWORDS CRATE PATH)
+    set(MULTI_VALUE_KEYWORDS FILES)
+    cmake_parse_arguments(_arg "${OPTIONS}" "${ONE_VALUE_KEYWORDS}" "${MULTI_VALUE_KEYWORDS}" ${ARGN})
+
+    if("${_arg_CRATE}" STREQUAL "")
+        message(FATAL_ERROR "corrosion_cxxbridge called without a crate")
+    endif()
+
+    if("${_arg_PATH}" STREQUAL "")
+        message(FATAL_ERROR "corrosion_cxxbridge called without path to folder of Cargo.toml of target ${_arg_CRATE}")
+    endif()
+
+    if(NOT EXISTS "${CMAKE_CURRENT_LIST_DIR}/${_arg_PATH}/Cargo.toml")
+        message(FATAL_ERROR "The path ${CMAKE_CURRENT_LIST_DIR}/${_arg_PATH} does not contain a Cargo.toml")
+    endif()
+
+    if("${_arg_FILES}" STREQUAL "")
+        message(FATAL_ERROR "Must supply at least the path of a single file to be bridged, defined as relative path from Cargo.toml")
+    endif()
+
+    set(crate_path ${_arg_PATH})
+
+    add_library(${target})
```

Comment:
Do we need to do the `add_library` here? How about if we let the user create the cxx target (library or executable).
At first glance, the following  `target_include_directories`, `target_link_libraries` and `target_sources` should work just as well on a user created target (and thus would also work for C++ executables). Am I missing something?

Reply:
Hmm, unsure. I could test and see how it works. I do think it makes sense for this to create a whole target and not sure the actual benefit of the user creating its own target just to link this in. But I am likely missing something also.
If there is not a problem I can do as you wish.

Reply:
Currently your example is:

> ```cmake
> cmake_minimum_required(VERSION 3.24)
> project(example CXX)
> 
> find_package(corrosion REQUIRED)
> corrosion_import_crate(MANIFEST_PATH someCratePath/Cargo.toml)
> corrosion_cxxbridge(myCxxTarget CRATE someCrate PATH someCratePath FILES src/lib.rs)
> 
> add_executable(example main.cpp)
> target_link_libraries(example PRIVATE myCxxTarget)
> ```

But I think it could be:

```cmake
cmake_minimum_required(VERSION 3.24)
project(example CXX)

find_package(corrosion REQUIRED)
corrosion_import_crate(MANIFEST_PATH someCratePath/Cargo.toml)

add_executable(example main.cpp) # The user will anyway create this target, no need for intermediate C library
corrosion_cxxbridge(example CRATE someCrate PATH someCratePath FILES src/lib.rs)
```

This way the user can choose themselves if they want to integrate the cxx bindings into a library or directly into an executable, without an intermediate C library.

Reply:
We could. But I do think there is value in constraining the ways to do it. I think the current way is flexible enough to do anything they need to do, but inflexible enough to limit things that can create "spaghetti", both for our and the users sake.
I don't think it matters much however.

We could rename the function to `corrosion_add_cxxbridge` which then makes it in line with `add_library`?

Reply:
> We could rename the function to corrosion_add_cxxbridge which then makes it in line with add_library?

I'd be happy with that.